CommunicationLayer/CommDeviceControl/InternetConnectionService.cpp

```diff
@@ -0,0 +1,82 @@
+#include <SimpleAmqpClient/SimpleAmqpClient.h>
+#include <QUdpSocket>
+#include <QDebug>
+#include <QTextStream>
+#include <QTimer>
+
+#include "../../InfrastructureLayer/Settings/I_Settings.h"
+#include "InternetConnectionService.h"
+namespace
+{
+    quint32 SLEEP_TIME_MILLISECONDS = 2000;
+}
+
+InternetConnectionService::InternetConnectionService(
+    QString exchangeName
+    ,  QString ipAddress
+    ,  quint16 port)
+    : exchangeName_(exchangeName)
+    , ipAddress_(ipAddress)
+    , port_(port)
+    , connectionRetryTimer_(new QTimer(this))
+{
+    QObject::connect(this, SIGNAL(setupChannelSignal()), this, SLOT(setupChannel()));
+
+    //connectionRetryTimer_& = new QTimer(this);
+    connectionRetryTimer_.setSingleShot(true);
+    connect(&connectionRetryTimer_, SIGNAL(timeout()), this, SLOT(setupChannel()));
+    connectToDataSource();
+}
+
+InternetConnectionService::~InternetConnectionService()
+{
+}
+
+
+void InternetConnectionService::setupChannel()
+{
+    qWarning() << "InternetConnectionService: Attempting to connect to " << ipAddress_ << port_;
+
+    try
+    {
+        channel_ = AmqpClient::Channel::Create(ipAddress_.toStdString(), port_);
+    }
+    catch (AmqpClient::AmqpException::exception&)
```

Comment:
Can this get more granular? There should be custom error messages depending on what specific kind of error.